sklearn/naive_bayes.py

```diff
@@ -117,6 +115,148 @@ def predict_proba(self, X):
         return np.exp(self.predict_log_proba(X))
 
 
+class GeneralNB(_BaseNB):
+    """General Naive Bayes
+
+    Read more in the :ref:`User Guide <general_naive_bayes>`.
+
+    Attributes
+    ----------
+    fits_ : list of objects
+        list of fitted classifiers
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> X = np.array([[1.5, 2.3, 5.7, 0, 1],
+    ...               [2.7, 3.8, 2.3, 1, 0],
+    ...               [1.7, 0.1, 4.5, 1, 0]])
+    >>> y = np.array([1, 0, 0])
+    >>> from sklearn.naive_bayes import GeneralNB, GaussianNB, BernoulliNB
+    >>> clf = GeneralNB()
+    >>> clf.fit(X, y, [(GaussianNB(), [0, 1, 2]),
+    ...                (BernoulliNB(), [3, 4])])
+    GeneralNB()
+    >>> print(clf.predict([[1.5, 2.3, 5.7, 0, 1]]))
+    [1]
+    >>> print(clf.score([[2.7, 3.8, 1, 0, 1]],[0]))
+    1.0
+    """
+
+    def __init__(self):
+        self.fits_ = []
+
+    def fit(self, X, y, distributions):
+        """Fit Gaussian Naive Bayes according to X, y
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape (n_samples,)
+            Target values.
+        distributions : list of tuples
```

Comment:
I think this needs to be a parameter to the constructor of the class.
It might be best to replicate the syntax (and some of the implementation) of ColumnTransformer for specifying estimators for dividing the work by columns. There we have transformers, here probability models.

Reply:
Got it, I'll have a look at it.

Reply:
@jnothman, the GeneralNB now subclasses `_BaseNB`, `_BaseComposition` and `ClassifierMixin`. The constructor takes in a list of tuples which was previously in the `fit()` method. The attributes and methods are heavily inspired from the ColumnTransformer.

Reply:
Thanks! Drop this docstring then?

Reply:
Fixed.